### ISSUE TITLE
Using multiple stylesheets per block does not work in Site Editor

### ISSUE THREAD

**Issue by gyurmey2** [WordPress/gutenberg#46702]
### Description

According to [this manual](https://make.wordpress.org/core/2021/12/15/using-multiple-stylesheets-per-block/), styles are only loaded in the Page Editor and Frontend, but not in the Site Editor.

### Step-by-step reproduction instructions

**1.** Add the following code snippet to the functions.php file:

```
add_action('after_setup_theme', function () {

  $args = array(
    'handle' => 'my-theme-paragraph',
    'src'    => get_theme_file_uri('assets/blocks/paragraph.css'),
  );

  $args['path'] = get_theme_file_path('assets/blocks/paragraph.css');

  wp_enqueue_block_style('core/paragraph', $args);
});
```

**2.** Add styles in the paragraph.css file:

`p { background: red}`

**3.** Check that the styles are loaded in the Site Editor.

### Front-end:
![frontend](https://user-images.githubusercontent.com/113413673/208888749-167b7fc5-c5da-49ad-975b-4cb7e2bcadb9.png)

### Page Editor:
![page-editor](https://user-images.githubusercontent.com/113413673/208888506-d5a2ce5d-ba4b-4672-ad8a-8e03882bbb01.png)

### Site Editor:
![site-editor](https://user-images.githubusercontent.com/113413673/208888541-701d2b44-128c-4966-ab3e-a642d3452e33.png)

### Environment info

Fresh install WP (6.1.1) with TT3 theme.
Gutenberg: 14.7.3 & 14.9.20221216 & without.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by kathrynwp** [comment]
@gyurmey2 Thank you for the report!

@aristath I'm taking the liberty of looping you in here, in case you have a moment to take a peek at this. Thanks!

**Comment by MadtownLems** [comment]
I believe the issue here is that - in the Site Editor - the block is rendered inside of an iframe that does not have actually have the CSS file included. The CSS file is only in the top-level frame, which is why the paragraph in the sidebar is red.

I think there are lots of open issues related to this, such as: https://github.com/WordPress/gutenberg/issues/41821

**Comment by ndiego** [comment]
I am able to confirm this, but I could have sworn this used to work. @justintadlock I know you recently wrote an article on per-block stylesheets. Did you run into this?

**Comment by justintadlock** [comment]
> I am able to confirm this, but I could have sworn this used to work. @justintadlock I know you recently wrote an article on per-block stylesheets. Did you run into this?

Actually, I didn't test loading multiple stylesheets per block.  But, loading a single stylesheet per block is working for me:

![custom-block-stylesheet](https://user-images.githubusercontent.com/1816309/212191443-70371204-16d1-4621-8487-bc1a1300a31c.jpg)

I'm also registering on `init`.

**Comment by MadtownLems** [comment]
I don't remember where I saw this, but I believe that the issue in the original post is that the stylesheet doesn't contain any rules about "wp-block".

I just made a sample plugin with the following CSS only, registered with admin_enqueue_scripts, and it does NOT get included in the Site Editor:
.j-sample-block-output-backend-red p { color: red; }


however, only changing the CSS file to add a line containing .wp-block makes the CSS be included in the Site Editor:

.j-sample-block-output-backend-red p { color: red; }

.wp-block { background-color: yellow; }

**Comment by MadtownLems** [comment]
This might be the best thread for discussion around the need for "wp-block" to be included in the CSS file:
https://github.com/WordPress/gutenberg/pull/40842

**Comment by pagelab** [comment]
Adding a random `wp-block` class worked for me, but the “plain” paragraph style keeps leaking into the sidebar. 

<img width="419" alt="Screen-Shot-2023-01-16-15-20-43" src="https://user-images.githubusercontent.com/1451087/212744210-e2c298eb-8474-40e1-a5a9-ca13957bd1aa.png">

Maybe the `.editor-styles-wrapper` class should be automatically included in the editor for stylesheets enqueued with `wp_enqueue_block_style`.

**Comment by ndiego** [comment]
Yes, I can confirm @pagelab's results. Adding a per-block stylesheet for paragraphs without `wp-block` in the file loads the CSS in the Editor but not in the iframe. This leads to paragraphs in the inspector being styled but not the paragraphs in the actual page content. 

![image](https://user-images.githubusercontent.com/4832319/212917688-5e8f74ff-0c49-43a4-833b-e696a0ddcf46.png)

I added the following to a paragraph stylesheet:

```
.wp-block {}
p { background: red}
```

This then causes the file to be loaded in the iframe, but the styles also bleed into the Site Editor. 

![image](https://user-images.githubusercontent.com/4832319/212917495-50dff517-b70b-4041-91b4-011d5022099b.png)

**Comment by ellatrix** [comment]
This should be fixed by #49655 (in WP 6.3), because `wp_enqueue_block_style` is internally using `enqueue_block_assets`.

**Comment by alogonzac** [comment]
I am still experiencing this issue on version 6.4.2
